src/Common/tests/System/Xml/XPath/XPathExpressionTests/EvaluateTests.cs

```diff
@@ -195,4 +198,44 @@ public static void Variation_12()
             Assert.Throws<XPathException>(() => navigator.Select(String.Empty));
         }
     }
+
+    public class XPathEvaluateTests
+    {
+        [Fact]
+        public static void EvaluateTextNode_1()
+        {
+            XElement element = XElement.Parse("<element>Text.</element>");
+            IEnumerable result = (IEnumerable)element.XPathEvaluate("/text()");
+            Assert.Equal(1, result.Cast<XText>().Count());
+            Assert.Equal("Text.", result.Cast<XText>().First().ToString());
+        }
+
+        [Fact]
+        public static void EvaluateTextNode_2()
+        {
+            XElement element = XElement.Parse("<root>1<element></element>2</root>");
+            IEnumerable result = (IEnumerable)element.XPathEvaluate("/text()[1]");
+            Assert.Equal(1, result.Cast<XText>().Count());
+            Assert.Equal("1", result.Cast<XText>().First().ToString());
+        }
+
+        [Fact]
+        public static void EvaluateTextNode_3()
+        {
+            XElement element = XElement.Parse("<root>1<element></element>2</root>");
+            IEnumerable result = (IEnumerable)element.XPathEvaluate("/text()[2]");
+            Assert.Equal(1, result.Cast<XText>().Count());
+            Assert.Equal("2", result.Cast<XText>().First().ToString());
+        }
+
+        [Fact]
+        public static void EvaluateTextNode_4()
+        {
+            XElement element = XElement.Parse("<root>1<element>2</element><element>3</element>4</root>");
+            IEnumerable result = (IEnumerable)element.XPathEvaluate("/element/text()[1]");
```

Comment:
I was confused by this xpath but I realize [1] means "that are first child" as opposed to (/element/text())[1] which means "second of the text elements"